analyzers/src/SonarAnalyzer.Common/Rules/ExpressionComplexityBase.cs

```diff
@@ -49,6 +49,9 @@ public abstract class ExpressionComplexityBase<TExpression, TSyntaxKind> : Param
                 {
                     if (c.Node.Parent?.Kind() is TSyntaxKind kind && (ComplexityIncreasingKinds.Contains(kind) || TransparentKinds.Contains(kind)))
                     {
+                        // The parent of the expression is itself complexity increasing (e.g. &&) or transparent (e.g. parenthesis).
+                        // We are only interested in the expression roots so we only report once per expression tree. Therefore we ignore any inner children, e.g.:
+                        // a && b && c -> (Left: (Left: a, Right: b), Right: c) // We are only interested in the outer expression (the one with Right: c)
```

Comment:
in this example, I don't understand which is the node which is passed to initialize? 

and what's the relationship between  `Right: c` and `a && b && c`?

Reply:
(I clarified sync that I was confused of the usage of the comment syntax and thought the whole line had two expressions, one logical and one pattern-matched based)